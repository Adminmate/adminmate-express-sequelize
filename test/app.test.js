// see https://stackoverflow.com/questions/46227783/encoding-not-recognized-in-jest-js
require('mysql2/node_modules/iconv-lite').encodingExists('foo');

const supertest = require('supertest');
const jwt = require('jwt-simple');

// Include the app
const app = require('./app.js');

// Endpoint prefix
const prefix = '/adminmate/api';

// Generate the admin token
const adminToken = jwt.encode({
  exp_date: Date.now() + 1000
}, 'authkey_secret');

// expect(response.body.status).toBe('success');
// expect(response).toMatchSnapshot();
//  expect.objectContaining({
//   x: expect.any(Number),
//   y: expect.any(Number),
// })

// check_connection
describe('Testing POST /api/check_connection', () => {
  it('should return a 403 http response', async () => {
    // Add a little delay
    await new Promise((r) => setTimeout(r, 5000));

    // Make request
    const response = await supertest(app)
      .post(prefix + '/check_connection')
      .send({
        // data: ''
      });

    // Check response
    expect(response.status).toBe(403);
    expect(response.body.message).toBe('Invalid request');
  });
});

// login
describe('Testing POST /api/login', () => {
  it('should return a 403 http response', async () => {
    // Make request
    const response = await supertest(app)
      .post(prefix + '/login')
      .send({
        // data: ''
      });

    // Check response
    expect(response.status).toBe(403);
    expect(response.body.message).toBe('Invalid request');
  });
});

// Models
describe('Testing GET /api/models', () => {
  it('should return a 403 http response', async () => {
    // Make request
    const response = await supertest(app)
      .get(prefix + '/models');

    // Check response
    expect(response.status).toBe(403);
    expect(response.body.code).toBe('not_authorized');
  });

  it('should return a 200 http response', async () => {
    // Make request
    const response = await supertest(app)
      .get(prefix + '/models')
      .set('x-access-token', adminToken);

    // Check response
    expect(response.status).toBe(200);
    expect(response.body.models).toMatchSnapshot();
  });
});