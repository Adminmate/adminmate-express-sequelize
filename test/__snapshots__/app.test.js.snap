// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testing GET /api/models should return a 200 http response 1`] = `
Array [
  Object {
    "properties": Array [
      Object {
        "path": "id",
        "type": "Number",
      },
      Object {
        "path": "firstname",
        "type": "String",
      },
      Object {
        "path": "lastname",
        "type": "String",
      },
      Object {
        "path": "birthdate",
        "type": "Date",
      },
      Object {
        "path": "rating",
        "type": "Number",
      },
      Object {
        "path": "createdAt",
        "type": "Date",
      },
      Object {
        "path": "updatedAt",
        "type": "Date",
      },
    ],
    "segments": Array [],
    "slug": "users",
    "smart_actions": Array [],
  },
  Object {
    "properties": Array [
      Object {
        "path": "id",
        "type": "Number",
      },
      Object {
        "path": "name",
        "type": "String",
      },
      Object {
        "path": "model",
        "type": "String",
      },
      Object {
        "path": "year",
        "type": "Date",
      },
      Object {
        "path": "createdAt",
        "type": "Date",
      },
      Object {
        "path": "updatedAt",
        "type": "Date",
      },
      Object {
        "path": "userId",
        "ref": "users",
        "type": "Number",
      },
    ],
    "segments": Array [],
    "slug": "cars",
    "smart_actions": Array [],
  },
]
`;
